src/pubsub/mod_pubsub_db_mnesia.erl

```diff
@@ -200,11 +227,41 @@ del_node(Nidx) ->
 -spec set_node(mod_pubsub:pubsubNode()) -> {ok, mod_pubsub:nodeIdx()}.
 set_node(#pubsub_node{id = undefined} = Node) ->
     CreateNode = Node#pubsub_node{id = pubsub_index:new(node)},
-    set_node(CreateNode);
-set_node(#pubsub_node{id = Nidx} = Node) ->
+    OldParents = [],
+    set_node(CreateNode, OldParents);
+set_node(Node) ->
+    OldParents = get_parentnodes_names(Node),
+    set_node(Node, OldParents).
+
+set_node(#pubsub_node{id = Nidx} = Node, OldParents) ->
     mnesia:write(Node),
```

Comment:
is the `mnesia:transaction` wrapped required?

Reply:
Not required.
A caller is calling the function from inside a transaction to begin with.